app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuDataCollectionRobot.kt

```diff
@@ -0,0 +1,89 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ui.robots
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.Visibility
+import androidx.test.espresso.matcher.ViewMatchers.withContentDescription
+import androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withParent
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import org.hamcrest.CoreMatchers.allOf
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.assertIsEnabled
+import org.mozilla.fenix.helpers.click
+
+/**
+ * Implementation of Robot Pattern for the settings Data Collection sub menu.
+ */
+class SettingsSubMenuDataCollectionRobot {
+
+    fun verifyNavigationToolBarHeader() = assertNavigationToolBarHeader()
+
+    fun verifyDataCollectionOptions() = assertDataCollectionOptions()
+
+    fun verifyUsageAndTechnicalDataSwitchDefault() = assertUsageAndTechnicalDataSwitchDefault()
+
+    fun verifyMarketingDataSwitchDefault() = assertMarketingDataValueSwitchDefault()
+
+    fun verifyDataCollectionSubMenuItems() {
+        verifyDataCollectionOptions()
+        verifyUsageAndTechnicalDataSwitchDefault()
+        verifyMarketingDataSwitchDefault()
+    }
+
+    class Transition {
+        val mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())!!
+        fun goBack(interact: SettingsRobot.() -> Unit): SettingsRobot.Transition {
+            goBackButton().click()
+
+            SettingsRobot().interact()
+            return SettingsRobot.Transition()
+        }
+    }
+}
+
+private fun goBackButton() =
+    onView(withContentDescription("Navigate up"))
+
+private fun assertNavigationToolBarHeader() = onView(
+    allOf(withParent(withId(R.id.navigationToolbar)),
+        withText(R.string.preferences_data_collection)))
+    .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+
+private fun assertDataCollectionOptions() {
+
+    onView(withText(R.string.preference_usage_data))
+        .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+
+    val usageAndTechnicalDataText =
+        "Shares performance, usage, hardware and customization data about your browser with Mozilla to help us make Firefox Preview better"
+
+    onView(withText(usageAndTechnicalDataText))
+        .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+
+    onView(withText(R.string.preferences_marketing_data))
+        .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+
+    val marketingDataText =
+        "Shares data about what features you use in Firefox Preview with Leanplum, our mobile marketing vendor."

```

Comment:
Is not this string `R.string.preferences_marketing_data_description` working here?  And similar for the rest of string defined this way...

Reply:
Yes, they are not working.

While I was using (marketing data description)and (usage and technical description) I am facing no matching views found error.

Reply:
I guess that's because the %s  found there...

Reply:
Yes, that's exactly I was trying to convey in the other comment :p